### ISSUE TITLE
Woodpecker: fix private repositories

### ISSUE THREAD

**Issue by pinpox** [lounge-rocks/the-lounge#14]
In the currently deployed version of woodpecker (1.0.2) private repositories are broken. 
Current master (as of 2023-08-18) breaks the `clone` step for local backend on certain occations, a fix is on it's way here: https://github.com/woodpecker-ci/woodpecker/pull/2241

When that is merged, we should have a version that can do both (private repos and clone from local backend)

**Comment by 6543** [comment]
@pinpox `next-0c282e86e8` got just published witch should fix it

**Comment by MayNiklas** [comment]
> @pinpox `next-0c282e86e8` got just published witch should fix it
> 
> private repos (in local backend) where never supported for < 1.0.x ... so you do have to use next (upcomming v1.1.x)

Thank you!! 👍
That is something we should be able to work with!

**Comment by MayNiklas** [comment]
@pinpox Should we deploy https://github.com/woodpecker-ci/woodpecker/commit/0c282e86e8e9a477ea9ad52ee1c8348455189629 this evening? This commit should the next-0c282e86e8 pre-release.

**Comment by 6543** [comment]
? Regression ?

**Comment by MayNiklas** [comment]
> ? Regression ?

Building the Woodpecker server (the Nix Package) failed because of a Node error.
https://drone.lounge.rocks/lounge-rocks/the-lounge/208/2/4

We decided to wait for the full release.

**Comment by 6543** [comment]
drone: 404 :/

**Comment by MayNiklas** [comment]
> drone: 404 :/

Ahh - our Drone server uses GeoIP to only allow countries we expect requests from 😅

**Comment by MayNiklas** [comment]
```log
building '/nix/store/wnrj7dziv68jagqs0kwvh8nbnyvdg221-woodpecker-agent-0c282e86e8e9a477ea9ad52ee1c8348455189629-go-modules.drv'...

building '/nix/store/756vw7chkqvgm3c9zvhiarlh719rpvqr-woodpecker-ci-0c282e86e8e9a477ea9ad52ee1c8348455189629.drv'...

error: builder for '/nix/store/756vw7chkqvgm3c9zvhiarlh719rpvqr-woodpecker-ci-0c282e86e8e9a477ea9ad52ee1c8348455189629.drv' failed with exit code 1;

       last 10 log lines:

       >   errno: -2,

       >   code: 'ENOENT',

       >   syscall: 'copyfile',

       >   path: 'node_modules/javascript-time-ago/locale/de.json.js',

       >   dest: 'src/assets/timeAgoLocales/de.js'

       > }

       >

       > Node.js v18.17.1

       > error Command failed with exit code 1.

       > info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

       For full logs, run 'nix log /nix/store/756vw7chkqvgm3c9zvhiarlh719rpvqr-woodpecker-ci-0c282e86e8e9a477ea9ad52ee1c8348455189629.drv'.

error: 1 dependencies of derivation '/nix/store/caah1smci2jb70v3abg364j87k8il13f-woodpecker-server-0c282e86e8e9a477ea9ad52ee1c8348455189629.drv' failed to build

error: 1 dependencies of derivation '/nix/store/hpnkmdvkv52fvq9v7wqa09hv35x671n5-unit-woodpecker-server.service.drv' failed to build

error (ignored): error: cannot unlink '/tmp/nix-build-woodpecker-agent-0c282e86e8e9a477ea9ad52ee1c8348455189629-go-modules.drv-0/go/pkg/mod/golang.org/x/tools@v0.7.0/internal/event/export': Directory not empty

error: 1 dependencies of derivation '/nix/store/3jgwvi3qz56agflc4s2vyvbm4yjj82lv-system-units.drv' failed to build

error: 1 dependencies of derivation '/nix/store/anj5fy9y4p096yz2zgmwxm3dlyib0ilv-etc.drv' failed to build

error: 1 dependencies of derivation '/nix/store/sy4vwbmhschdgbfvckvy694zy91csrqs-nixos-system-woodpecker-server-23.11.20230816.caac0eb.drv' failed to build
```

**Comment by pinpox** [comment]
@6543 This the full error when building. But I think it's nix-related, probably something for @ambroisie to fix on the next release if persists, as he is the maintainer of the package and module in nixpkgs.

```
❯ nix log /nix/store/lvyykarlva5smyy1n97is6fhqw31cqmj-woodpecker-ci-0c282e86e8e9a477ea9ad52ee1c8348455189629.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/lvyykarlva5smyy1n97is6fhqw31cqmj-woodpecker-ci-0c282e86e8e9a477ea9ad52ee1c8348455189629.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/spb4acn3p3rvzzp480xgk084cqyqhigg-source/web
source root is web
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
yarn run v1.22.19
warning Skipping preferred cache folder "/homeless-shelter/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/build/.yarn-cache-1000".
$ vite build --base=/BASE_PATH
warning Cannot find a suitable global folder. Tried these: "/usr/local, /homeless-shelter/.yarn"
/build/web/deps/woodpecker-ci/vite.config.ts:89
                throw err;
                ^

[Error: ENOENT: no such file or directory, copyfile 'node_modules/javascript-time-ago/locale/de.json.js' -> 'src/assets/timeAgoLocales/de.js'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'copyfile',
  path: 'node_modules/javascript-time-ago/locale/de.json.js',
  dest: 'src/assets/timeAgoLocales/de.js'
}

Node.js v18.17.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by ambroisie** [comment]
I _believe_ this is an intermittent error, as `nixpkgs-review` built the packges successfully.

**Comment by pinpox** [comment]
Related: https://github.com/pinpox/woodpecker-flake-pipeliner/issues/1

**Comment by MayNiklas** [comment]
Got fixed with 2.3.0
